app/_hub/kong-inc/openid-connect/how-to/_demonstrating_proof_of_possession.md

```diff
@@ -0,0 +1,103 @@
+
+It's possible to use Demonstrating Proof-of-Possession without mTLS, or even with a plain HTTP. But HTTPS is recommended for better protection.
+
+Note that by enabling the verification of the DPoP proof, Kong removes the `DPoP` header and changes the token type from "dpop" to "bearer", which essentially downgrades the request to use a conventional bearer token.
+This allows an OAuth2 upstream without DPoP support to work with the DPoP token without loosing the protection of key-binding mechanism.
+
+It's a prerequisite that the auth server is configured to enable the DPoP for you to enable it with Kong.
+
+If you are configuring this in Keycloak, see the [Keycloak configuration](#prerequisites) section in the prerequisites.
+For alternative auth servers, consult their documentation to configure this functionality.
+
+Some of the instructions in the other how-to guides for OpenID Connect support validation of access tokens using mTLS proof of possession.
+Enabling the [`proof_of_possession_dpop`](/hub/kong-inc/openid-connect/configuration/#config-proof_of_possession_dpop) configuration option in the plugin helps to ensure that the supplied access token
+belongs to the client by verifying its binding with the client certificate provided in the request.
+
+The DPoP are supported by the following auth methods:
+
+- [JWT Access Token Authentication](/hub/kong-inc/openid-connect/how-to/authentication/jwt-access-token/)
+- [Introspection Authentication](/hub/kong-inc/openid-connect/how-to/authentication/introspection/)
+- [Session Authentication](/hub/kong-inc/openid-connect/how-to/authentication/session/)
+
+   Session Authentication is only compatible with DPoP when used along with one of the other supported authentication methods:
+
+    * If multiple `openid-connect` plugins are configured with the `session` auth method, we strongly recommend configuring different values of [`session_secret`](/hub/kong-inc/openid-connect/configuration/#config-session_secret) across plugin instances for additional security. This avoids sessions being shared across plugins and possibly bypassing the proof of possession validation.
+
+The following example shows how to enable this feature for the JWT Access Token Authentication method. Similar steps can be followed for the other methods.
+
+## Demo
+### Prerequisites
+
+Follow these prerequisites to set up a demo Keycloak app and a Kong service and route for testing mTLS client auth.
+
+{% include_cached /md/plugins-hub/oidc-prereqs.md %}
+
+### Configure Demonstrating Proof-of-Possession
+
+1. To enable Demonstrating Proof-of-Possession, use the [`proof_of_possession_dpop`](/hub/kong-inc/openid-connect/configuration/#config-proof_of_possession_dpop) configuration option:
+
+    ```bash
+    http -f put :8001/plugins/5f35b796-ced6-4c00-9b2a-90eef745f4f9 \
```

Comment:
Can you convert this to cURL? We try to avoid using httpie since it's a separate tool that someone would have to install, whereas we can assume they have curl.